dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java

```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
+                .contentTypeId(blogContentType.id())
+                .values(String.valueOf(RELATIONSHIP_CARDINALITY.MANY_TO_MANY.ordinal()))
+                .relationType(commentContentType.variable()).required(false).build();
+
+        contentTypeFieldAPI.save(field, user);
+
+        //Creates parent content
+        Contentlet blogContentParent = new ContentletDataGen(blogContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
+                .setProperty("sysPublishDate", new Date())
+                .setProperty("body", "blogBody").next();
+
+        //Creates child content
+        final Contentlet commentContentChild = new ContentletDataGen(commentContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "commentContentChild")
+                .setProperty("urlTitle", "commentContentChild")
+                .setProperty("author", "systemUser")
+                .setProperty("sysPublishDate", new Date())
+                .setProperty("body", "blogBody").nextPersisted();
+
+        //Adds a new relationship between both contentlets
+        final Relationship relationship = APILocator.getRelationshipAPI()
+                .byContentType(blogContentType).get(0);
+
+        ContentletRelationships contentletRelationships = new ContentletRelationships(
+                blogContentParent);
+        ContentletRelationshipRecords records = contentletRelationships.new ContentletRelationshipRecords(
+                relationship, true);
+        records.setRecords(Lists.newArrayList(commentContentChild));
+        contentletRelationships.getRelationshipsRecords().add(records);
+        blogContentParent = contentletAPI.checkin(blogContentParent, contentletRelationships, null, null, user, false);
+
+        //Validate results from parent side
+        List<Contentlet> results = relationshipAPI.dbRelatedContent(relationship, blogContentParent);
+        validateDbRelatedContentResults(commentContentChild, results);
+
+        //Validate results from child side
+        results = relationshipAPI.dbRelatedContent(relationship, commentContentChild);
+        validateDbRelatedContentResults(blogContentParent, results);
+    }
+
+    /**
+     * <b>Method to test:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} <p>
+     * <b>Given Scenario:</b> A content is related to another (same content type/self-join relationship) <p>
+     * <b>ExpectedResult:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} should always return related content
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void testDbRelatedContentWhenSelfJoinRelationship()
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)